src/main/java/edu/kis/powp/jobs2d/command/ICompoundCommand.java

```diff
@@ -1,12 +1,21 @@
 package edu.kis.powp.jobs2d.command;
 
+import java.util.ArrayList;
 import java.util.Iterator;
+import java.util.List;
 
 /**
  * Interface extending Job2dDriverCommand to execute more than one command.
  */
-public interface ICompoundCommand extends DriverCommand {
+public interface ICompoundCommand extends DriverCommand, Iterable<DriverCommand> {
 
-	public Iterator<DriverCommand> iterator();
+	Iterator<DriverCommand> iterator();
 
+	@Override
+	default DriverCommand clone() throws CloneNotSupportedException{
+		List<DriverCommand> commands = new ArrayList<>();
+		this.forEach(commands::add);
```

Comment:
I am not sure that this is enough. You only copy references to existing commands (without making a copy). This not how cloning should look.